src/pk_encryption_circuit.rs

```diff
@@ -0,0 +1,753 @@
+            .map(|pk0_qi| PolyAssigned::new(ctx, pk0_qi))
+            .collect::<Vec<_>>();
+
+        let pk1_qi = self
+            .pk1_qi
+            .iter()
+            .map(|pk1_qi| Poly::<F>::new(pk1_qi.clone()))
+            .collect::<Vec<_>>();
+        let pk1_qi_assigned = pk1_qi
+            .into_iter()
+            .map(|pk1_qi| PolyAssigned::new(ctx, pk1_qi))
+            .collect::<Vec<_>>();
+
+        let u = Poly::<F>::new(self.u.clone());
+        let u_assigned = PolyAssigned::new(ctx, u);
+
+        let e0 = Poly::<F>::new(self.e0.clone());
+        let e0_assigned = PolyAssigned::new(ctx, e0);
+
+        let e1 = Poly::<F>::new(self.e1.clone());
+        let e1_assigned = PolyAssigned::new(ctx, e1);
+
+        let k1 = Poly::<F>::new(self.k1.clone());
+        let k1_assigned = PolyAssigned::new(ctx, k1);
+
+        let r1is_assigned = self
+            .r1is
+            .iter()
+            .map(|r1is| {
+                let r1is = Poly::<F>::new(r1is.clone());
+                PolyAssigned::new(ctx, r1is)
+            })
+            .collect::<Vec<_>>();
+
+        let r2is_assigned = self
+            .r2is
+            .iter()
+            .map(|r2is| {
+                let r2is = Poly::<F>::new(r2is.clone());
+                PolyAssigned::new(ctx, r2is)
+            })
+            .collect::<Vec<_>>();
+
+        let p1is_assigned = self
+            .p1is
+            .iter()
+            .map(|p1is| {
+                let p1is = Poly::<F>::new(p1is.clone());
+                PolyAssigned::new(ctx, p1is)
+            })
+            .collect::<Vec<_>>();
+
+        let p2is_assigned = self
+            .p2is
+            .iter()
+            .map(|p2is| {
+                let p2is = Poly::<F>::new(p2is.clone());
+                PolyAssigned::new(ctx, p2is)
+            })
+            .collect::<Vec<_>>();
+
+        let ct0is_assigned = self
+            .ct0is
+            .iter()
+            .map(|ct0is| {
+                let ct0is = Poly::<F>::new(ct0is.clone());
+                PolyAssigned::new(ctx, ct0is)
+            })
+            .collect::<Vec<_>>();
+
+        let ct1is_assigned = self
+            .ct1is
+            .iter()
+            .map(|ct1is| {
+                let ct1is = Poly::<F>::new(ct1is.clone());
+                PolyAssigned::new(ctx, ct1is)
+            })
+            .collect::<Vec<_>>();
+
+        for pk0 in pk0_qi_assigned.iter() {
+            public_inputs.push(pk0.assigned_coefficients[0]);
+        }
+        for pk1 in pk1_qi_assigned.iter() {
+            public_inputs.push(pk1.assigned_coefficients[0]);
+        }
+        for ct0 in ct0is_assigned.iter() {
+            public_inputs.push(ct0.assigned_coefficients[0]);
+        }
+        for ct1 in ct1is_assigned.iter() {
+            public_inputs.push(ct1.assigned_coefficients[0]);
+        }
+
+        builder.base.assigned_instances[0] = public_inputs;
+
+        Payload {
+            pk0_qi_assigned,
```

Comment:
Why `pk0_qi` ? We always called it `pk0i`

Reply:
Not sure. Will correct.